lib/kvdag.rb

```diff
@@ -5,11 +5,29 @@
 require 'kvdag/edge'
 require 'kvdag/keypathhash'
 
+# Directed Acyclic Graph for multiple inheritance key-value lookup
+
 class KVDAG
   include Enumerable
+
+  # Return the set of all vertices
   attr_reader :vertices
+
   attr_reader :hash_proxy_class
 
+  # Create a new KVDAG, optionally using a specialized
+  # hash class for storing key-values. The default is to use
+  # a dot-separated keypath proxy for storing key-values like
+  #
+  # hsh["a.b.c"] = value
```

Comment:
rdoc doesn't seem to like this. I get the following HTML from it:
```html
<p><a href="http://" a.b.c""="">hsh</a> = value</p>
```
The rdoc [documentation](http://docs.seattlerb.org/rdoc/RDoc/Markup.html) says:
> Links can also be of the form label[url], in which case label is used in the displayed text, and url is used as the target.

You probably want to indent it, so it results in verbatim text.